teensy/pixy_main.ino

```diff
@@ -133,60 +137,108 @@ void loop()
-                    }
-                    else if (goal_angle > (185 + goal_threshold) && ball_found == false)
-                    {
-                        myMotors.MoveMotorsImu(270, abs(speed_t_goal), speed_w);
-                    }
-                    else
-                    {
-                        myMotors.MoveMotorsImu(ball_angle, 0, speed_w);
-                    }
-                }
     else
     {
-      Serial.println("No camera sees the ball");
-      myMotors.StopMotors();
+      //------------------ Angle normalization ------------------//
+      if (ball_angle < 180)
+      {
+        ball_angle_180 = -ball_angle;
+      }
+      else if (ball_angle > 180)
+      {
+        ball_angle_180 = 360 - ball_angle;
+      }
+      ball_angle_180 = ball_angle_180 * (-1);
+
+      //------------------ Camera detection cases ------------------//
+      if (ball_seen_pixy && ball_seen_openmv)
+      {
+        last_ball_angle = ball_angle;
+        if (distance_pixels < 85)
+        {
+          myMotors.MoveMotorsImu(goal_angle, speed_t_ball, speed_w);
+          esc.writeMicroseconds(mid_speed);
+          Serial.println("Both cameras see the ball and shoots");
+        }
+        else
+        {
+          myMotors.MoveMotorsImu(angle_degrees, speed_t_ball, speed_w);
+          Serial.println("Both cameras see the ball");
+        }
+      }
+      else if (ball_seen_pixy && !ball_seen_openmv)
+      {
+        last_ball_angle = ball_angle;
+        if (distance_pixels < 85)
+        {
+          myMotors.MoveMotorsImu(goal_angle, speed_t_ball, speed_w);
+          esc.writeMicroseconds(mid_speed);
+          Serial.println("Only pixy sees the ball and shoots");
+        }
+        else
+        {
+          myMotors.MoveMotorsImu(angle_degrees, speed_t_ball, speed_w);
+          Serial.println("Only pixy sees the ball");
+        }
+      }
+      else if (!ball_seen_pixy && ball_seen_openmv)
+      {
+        last_ball_angle = ball_angle;
+        double differential = ball_angle_180 * 0.15;
+        ponderated_angle = ball_angle + differential;
+        myMotors.MoveMotorsImu(ponderated_angle, abs(speed_t_ball), speed_w);
+        Serial.println("Only OpenMV sees the ball");
+      }
+      //---------------------- Center robot in goal ----------------------//
+      /*else if (goal_angle > 0)
+      {
+        if (goal_angle < (185 - goal_threshold) && ball_found == false)
+        {
+          myMotors.MoveMotorsImu(90, abs(speed_t_goal), speed_w);
+        }
+        else if (goal_angle > (185 + goal_threshold) && ball_found == false)
+        {
+          myMotors.MoveMotorsImu(270, abs(speed_t_goal), speed_w);
+        }
+        else
+        {
+          myMotors.MoveMotorsImu(ball_angle, 0, speed_w);
+        }
+      }*/
+     //-------------------- Move to last known position --------------------//
+      else
+      {
+        current_millis = millis();
+        if (current_millis - start_millis >= period)
```

Comment:
You are really smart for establishing an actuation `period`, you can play around with the value and see what works best

Reply:
Thanks!